### PR TITLE
DE-3800: Context.write_file(..., file_category="IDS") raises pylint error

### PR DESCRIPTION
**Pull request by VladimirShteyn**
## JIRA ticket
https://tetrascience.atlassian.net/browse/DE-3800

## Description
A static code analyzer that checks when `Context.write_file` is passed `file_category="IDS"`. It's robust to newlines since it parses the code's abstract syntax tree (AST). It's not robust to passing a variable whose value is "IDS" to `Context.write_file`. 
To do the above, I implemented a `pylint` plugin (`deprecation_checker.py`). (See [pylint docs for how to write a pylint plugin in general](https://pylint.pycqa.org/en/v2.13.9/how_tos/plugins.html).) A Github Action located in the `.github/actions/deprecation-checker` calls `pylint` using this Action.

## Supporting test data
* Unit tests
* Updated a Task Script such that it raises the `pylint` error. See [here](https://github.com/tetrascience/ts-task-script-molecular-devices-softmax-pro-raw-to-ids/runs/6472740157?check_suite_focus=true#step:4:136) for workflow run. In the [prior run](https://github.com/tetrascience/ts-task-script-molecular-devices-softmax-pro-raw-to-ids/runs/6472679999?check_suite_focus=true#step:4:117) no `pylint` error is raised because `Context.write_file` is passed `file_category=a_variable_name`.

## Automated testing

- [ ] Unit tests
- [ ] Integration tests
- [ ] API tests
- [ ] End-To-End tests
- [ ] N/A (please clarify)

### CODE REVIEW

File: .github/actions/deprecation-checker/deprecation_checker.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    :class:`ts_sdk.task.__task_script_runner.Context.write_file`.
+    """
+
+    # Note: the pylint plugin breaks without setting ``__implements__``.
+    __implements__ = (IAstroidChecker,)
+
+    name = "context-api"
+    #: Messages to register with pylint and potentially display.
+    msgs = {
+        "W1504": (
```
Comment (abclay) [inline .github/actions/deprecation-checker/deprecation_checker.py:41]: Just curious, how did you choose the number for this check?

File: .github/workflows/python_code_quality.yml [hunk 2]
```diff
@@ -16,58 +15,32 @@ on:
...
+          lfs: true
+      - name: Checkout custom pylint plugin
+        if: hashFiles('main.py') != ''
+        uses: actions/checkout@v3
         with:
-          python-version: ${{ inputs.python-version }}
-      # Check the files with black
-      - run: |
-          pip install black
-          black --check .
```
Comment (abclay) [inline .github/workflows/python_code_quality.yml:70]: I think we still want to check formatting.

**Review by abclay — COMMENTED** [review]
This is really cool!  Nice job.

I only have a couple questions/comments.

**Review by mlipschultz — COMMENTED** [review]
I don't see anything to add beyond what Alex commented on.

File: .github/actions/deprecation-checker/deprecation_checker.py (hunk 1 above)
Comment (VladimirShteyn) [reply .github/actions/deprecation-checker/deprecation_checker.py]: Changed to W1599.

**Review by abclay — APPROVED** [review]

File: .github/actions/deprecation-checker/deprecation_checker.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    $ pylint \
+        --disable=all \
+        --enable=deprecated-context-api \
+        --load-plugins=deprecation_checker \
+        --score=n \
+        test/raises_error_example.py
+
+    ************* Module test.raises_error_example
+    test/raises_error_example.py:1:0: W1504: Deprecated argument 'IDS' passed to Context.write_file() (deprecated-context-api)
+    test/raises_error_example.py:6:4: W1504: Deprecated argument 'IDS' passed to Context.write_file() (deprecated-context-api)
```
Comment (mlipschultz) [inline .github/actions/deprecation-checker/deprecation_checker.py:14]:
```suggestion
    test/raises_error_example.py:1:0: W1599: Deprecated argument 'IDS' passed to Context.write_file() (deprecated-context-api)
    test/raises_error_example.py:6:4: W1599: Deprecated argument 'IDS' passed to Context.write_file() (deprecated-context-api)
```

**Review by mlipschultz — APPROVED** [review]
Just one minor thing to correct, but otherwise looks good to me too

File: .github/actions/deprecation-checker/action.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
+name: "Deprecated API Scan"
```
[the remark this answers is not in the document]
Comment (VladimirShteyn) [reply .github/actions/deprecation-checker/action.yaml]: Updated.

**Pull request merged**